### PR TITLE
[SPARK-6397][SQL] Check the missingInput simply

### PR DESCRIPTION
**Pull request by watermen**

### CODE REVIEW

**Comment by haiyangsea** [comment]
It looks like a greate feature!

**Comment by marmbrus** [comment]
test this please

**Comment by watermen** [comment]
@liancheng @marmbrus Any more comment on this?

**Comment by liancheng** [comment]
LGTM, #5129 also needs this one. I'm merging this to master and branch-1.3. Thanks for working on this!

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala [hunk 1]
```diff
@@ -191,6 +191,8 @@ case class Expand(
     val sizeInBytes = child.statistics.sizeInBytes * projections.length
     Statistics(sizeInBytes = sizeInBytes)
   }
+
+  override def missingInput = super.missingInput.filter(_.name != VirtualColumn.groupingIdName)
```
Comment (liancheng) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala]: Why do we prefer adding `missingInput` in `Expand`? At first I though we should just update `QueryPlan.missingInput` to take the grouping ID virtual column into account.

**Comment by liancheng** [comment]
Sorry, just reverted my last merge, please see my comment above.

**Comment by liancheng** [comment]
@watermen A little off topic, would you mind to set your name properly by executing this:

```
$ git config --global user.name "Yadong Qi"
```

Currently your name shows as `q00251598` in the Git log. This can be inconvenient for us (mostly for @pwendell) to generate the credit list for the next release. Thanks!

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala (hunk 1 above)
Comment (watermen) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala]: @liancheng The grouping ID virtual column only used in Expand(LogicalPlan), so I think it's better to add the grouping ID virtual column in subclass(Expand) of QueryPlan.

**Comment by watermen** [comment]
@liancheng Thanks, I'll do it.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala (hunk 1 above)
Comment (liancheng) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala]: My point is we may also support other virtual columns like `INPUT__FILE__NAME` and `BLOCK__OFFSET__INSIDE__FILE` in the future. These virtual columns can be used in more general places like `Project` and `Filter`. That's why I think moving this to `QueryPlan` may make more sense.

**Comment by liancheng** [comment]
Since my previous merge operation closed this PR, @watermen has opened #5132 to supersede this.

**Pull request closed without merging**